lms/static/scripts/frontend_apps/components/dashboard/test/AssignmentActivity-test.js

```diff
@@ -392,6 +423,70 @@ describe('AssignmentActivity', () => {
     );
   });
 
+  context('when assignment has segments', () => {
+    it('sets no segments when auto-grading is not enabled', () => {
+      const wrapper = createComponent();
+      const filters = wrapper.find('DashboardActivityFilters');
+
+      assert.isUndefined(filters.prop('segments'));
+    });
+
+    [{}, { sections: [] }, { groups: [] }].forEach(assignmentExtra => {
+      it('sets no segments when assignment has no groups nor sections', () => {
+        setUpFakeUseAPIFetch({
+          ...activeAssignment,
+          ...assignmentExtra,
+          auto_grading_config: {},
+        });
+
+        const wrapper = createComponent();
+        const filters = wrapper.find('DashboardActivityFilters');
+
+        assert.isUndefined(filters.prop('segments'));
+      });
+    });
+
+    [
+      {
+        assignmentExtra: { sections: [{}, {}] },
+        expectedType: 'sections',
+      },
+      {
+        assignmentExtra: { groups: [{}, {}, {}] },
+        expectedType: 'groups',
+      },
+    ].forEach(({ assignmentExtra, expectedType }) => {
+      it('sets the right type of segment based on assignment shape', () => {
```

Comment:
```suggestion
      it('sets type of segment based on assignment data fields', () => {
```

I presume by "shape" you meant which type the assignment data corresponds to, but I think this may be unclear to a future reader.